promql/engine_test.go

```diff
@@ -634,21 +648,25 @@ load 10s
 			Start: time.Unix(10, 0),
 		},
 		{
-			Query:      "metric[20s:5s]",
-			MaxSamples: 3,
+			Query:        "sum_over_time(metric[10s:1s])",
```

Comment:
This is still a simple subquery, we need to also have some complexity inside the subquery itself.

Reply:
Is `sum(count_over_time(sum_over_time(metric[10s:1s])[10s:]))[10s:2s]` satisfactory?

Reply:
Yeah, we should probably test multiple levels of subqueries. Let's test the intermediates too, otherwise it'd be a bit much to mentally check all together.

Reply:
Sorry, but could you tell more on this? `multiple levels of subqueries` and `intermediates`.

Do they mean break this query test into multiple small tests as well and join them again as tests so that we can easily understand the result?

Reply:
That is a 3-level subquery, so it has multiple levels. Each node on the AST is an intermediates.

Yes, one per level should do it.

Reply:
Done! @brian-brazil